code/datums/actions/item_action.dm

```diff
@@ -105,3 +105,28 @@
 /datum/action/item_action/aim_mode/action_activate()
 	var/obj/item/weapon/gun/I = target
 	I.toggle_auto_aim_mode(owner)
+
+
+/datum/action/item_action/toggle_hydro
+	/// This references the TL84 flamer
+	var/obj/item/weapon/gun/flamer/marinestandard/holder_flamer
+
+/datum/action/item_action/toggle_hydro/New()
+	. = ..()
+	holder_flamer = holder_item
```

Comment:
There should be a holder_flamer = null in this datums Destroy proc to avoid referencing it when it is deleted, you can read more about that here http://www.byond.com/docs/ref/#/DM/garbage and by asking around